File: 0504-base-7/0504-base-7.cpp
class Solution {
public:
    string convertToBase7(int num) {
        if(num == 0){
            return "0";
        }
        string ans;
        bool flag = false;
        if(num < 0){
            flag = true;
            num *= -1;
        }
        while(num > 0){
            int n = num % 7;
            num /= 7;
            ans = to_string(n) + ans;
        }
        if(flag){
            ans = "-" + ans;
        }
        return ans;
    }
};

File: 0504-base-7/0504-base-7.cs
public class Solution {
    public string ConvertToBase7(int num) {
        //  https://leetcode.com/problems/base-7/solutions/3566784/simple-solution-c-100-efficient
        if(num > -7 && num < 7){
            return num.ToString();
        }

        StringBuilder sb = new StringBuilder();
        int n = Math.Abs(num);
        int rem = 0;
        while(n > 0){
            rem = n % 7;
            n /= 7;
            sb.Insert(0, rem);
        }

        if(num < 0){
            sb.Insert(0, "-");
        }
        return sb.ToString();
    }
}